plugin/auto_mkdir2.vim

```diff
@@ -24,11 +27,14 @@ endif
 "##########################################################
 " Commands
 command! -nargs=* -complete=dir MkdirP call s:mkdir_p(<q-args>, 1)
+command! -nargs=* -complete=dir MkdirPC call s:mkdir_p(<q-args>, v:cmdbang)
```

Comment:
I'm not entirely sure if I follow why this command is needed? Can't we just add `-bang` to `:MkdirP`? What is "C"?

You also need to pass `-bang` to `:command` for it to work btw, and `v:cmdbang` only works inside autocmds (to check if `:e!` was used); for commands you need to use `<bang>`, which expands to a `'!'` or `''`; the usual way to use it is:

    command! -bang Command :call s:fun(<bang>0)

Which will call either `s:fun(0)` or `s:fun(!0)` (passing `0` or `1` to the function).

Reply:
Adding `-bang` to `:MkdirP` works when it's used independently, but I wasn't able to figure out how to pass a `!` through a user-defined `autocmd`. I wanted to be able to `:w!` and override confirmation, so I guess I cheated and made a command that could only be used in `autocmds`. 

My knowledge of vimscript is just enough to hack out the changes I made initially. I learned a bunch just from your comments. Much appreciated

Reply:
> I wanted to be able to :w! and override confirmation, so I guess I cheated and made a command that could only be used in autocmds.

That's already how it should work, I think? With the current master:

    $ vim newdir/asd
    Create directory `newdir' [y/N]?
    N
    "newdir/asd" E212: Can't open file for writing
    :w!
    "newdir/asd" [New] 0L, 0B written

Reply:
Wait, I see your problem now: because the default `:autocmd` uses the script-local `s:mkdir_p()` function, you can't call this from your autocmd in your Vimrc, so you need this to get the same effect.

I'll make some updates which makes this easier.

Reply:
Exactly. I don't know what the best way to do this is. As you pointed out, being able to pass a variable to the `autocmd` in the script would be ideal, but I couldn't figure out a way to do that without using `exec`